llms/mlx_lm/utils.py

```diff
@@ -150,20 +189,27 @@ def generate(
         print("=" * 10)
         print("Prompt:", prompt)
 
-    prompt = mx.array(tokenizer.encode(prompt))
+    prompt_tokens = mx.array(tokenizer.encode(prompt))
+    repetition_context = []
+    repetition_context.extend(tokenizer.encode(prompt))
 
     tic = time.perf_counter()
     tokens = []
     skip = 0
     REPLACEMENT_CHAR = "\ufffd"
 
-    for (token, prob), n in zip(generate_step(prompt, model, temp), range(max_tokens)):
-        if token == tokenizer.eos_token_id:
+    for (token, prob), n in zip(
+        generate_step(
+            prompt_tokens, model, temp, repetition_penalty, repetition_context
+        ),
+        range(max_tokens),
+    ):
+        if token[0] == tokenizer.eos_token_id:
             break
         if n == 0:
             prompt_time = time.perf_counter() - tic
             tic = time.perf_counter()
-        tokens.append(token.item())
+        tokens.append(token[0].item())
```

Comment:
Issue: Changing `token` to `token[0]` seems unnecessary, and using `token[0]` as a magic number would be bad for readability. Could you please revert the changes?